src/check-reserve-update-frequency/check-reserve-update-frequency.spec.js

```diff
@@ -0,0 +1,125 @@
+  function mockTokensAddressesContractPromise(mockTimestamp) {
+    // create an array of reserve tokens
+    const reserveTokens = [
+      { symbol: 'FAKE1', tokenAddress: '0xFIRSTFAKETOKENADDRESS' },
+    ];
+
+    // create an array of contract addresses that correspond to the reserve tokens
+    const contractAddresses = [
+      '0xFIRSTFAKECONTRACTADDRESS',
+    ];
+
+    // create the Array of Tuples of token addresses / contract addresses / contract objects
+    // pass our mockTimestamp in to the mocked return value (roundData.updatedAt) to facilitate
+    // testing against different timestamp conditions
+    const tokenAddressesContractTuples = reserveTokens.map((token, index) => {
+      const address = contractAddresses[index];
+      // need to create mock contract that has .latestRoundData method and returns roundData
+      const contract = {
+        latestRoundData: jest.fn(() => Promise.resolve({
+          updatedAt: new BigNumber(mockTimestamp),
+        })),
+      };
+      return [token, address, contract];
+    });
+
+    // wrap the Array of Tuples in a Promise (which we will resolve immediately)
+    // the original function returns a Promise that is resolved when all contract interactions have
+    // finished
+    return Promise.resolve(tokenAddressesContractTuples);
+  }
+
+  describe('Reserve Price Oracle Monitoring', () => {
+    it('returns findings if price oracle age is more than 24 hours old', async () => {
+      // create a block timestamp and updatedAt timestamp that will trigger an alert
+      const blockTimestamp = new BigNumber(1234567890);
+      const oracleAgeTooOld = new BigNumber((24 * 60 * 60) + 1);
+      const updatedAtTimestamp = blockTimestamp.minus(oracleAgeTooOld);
+
+      // need to create blockEvent (with .block.timestamp and .blockNumber)
+      const mockedBlockEvent = createBlockEvent({
+        blockNumber: 12345,
+        block: {
+          timestamp: blockTimestamp,
+        },
+      });
+
+      // create the mocked promise to pass in to provideHandleBlock
+      const promise = await mockTokensAddressesContractPromise(updatedAtTimestamp);
+
+      // create the block handler
+      handleBlock = provideHandleBlock(promise);
+
+      // create expected finding
+      const token = promise[0][0];
+      const priceSourceAddress = promise[0][1];
+      const expectedFinding = Finding.fromObject({
+        name: `Stale AAVE Price Oracle Data for ${token.symbol}`,
+        description:
+        `Token ${token.symbol} Price Oracle Age: ${oracleAgeTooOld.toString()} seconds`,
+        alertId: 'AE-AAVE-PRICE-ORACLE-STALE',
+        severity: FindingSeverity.Medium,
+        type: FindingType.Suspicious,
+        everestId: '0xa3d1fd85c0b62fa8bab6b818ffc96b5ec57602b6',
+        metadata: {
+          symbol: token.symbol,
+          tokenAddress: token.tokenAddress,
+          oracleAge: oracleAgeTooOld,
+          priceSourceAddress,
+        },
+      });
```

Comment:
Something I like doing is moving this into a shared location or even importing it from the agent itself. Keeps code dupe down and makes it so you only have to update 1 spot if we update this in the future /shrug

Reply:
I've removed the duplicated code and added an import of `createAlert()` in the `.spec.js` file.